lib/c-api/src/wasm_c_api/engine.rs

```diff
@@ -269,6 +270,45 @@ pub extern "C" fn wasm_config_set_engine(config: &mut wasm_config_t, engine: was
     config.engine = engine;
 }
 
+/// Updates the configuration to enable NaN canonicalization.
+///
+/// This is a Wasmer-specific function.
+///
+/// # Example
+///
+/// ```rust
+/// # use inline_c::assert_c;
+/// # fn main() {
+/// #    (assert_c! {
+/// # #include "tests/wasmer_wasm.h"
+/// #
+/// int main() {
+///     // Create the configuration.
+///     wasm_config_t* config = wasm_config_new();
+///
+///     // Enable NaN canonicalization.
+///     wasm_config_enable_nan_canonicalization(config);
+///
+///     // Create the engine.
+///     wasm_engine_t* engine = wasm_engine_new_with_config(config);
+///
+///     // Check we have an engine!
+///     assert(engine);
+///
+///     // Free everything.
+///     wasm_engine_delete(engine);
+///
+///     return 0;
+/// }
+/// #    })
+/// #    .success();
+/// # }
+/// ```
+#[no_mangle]
+pub extern "C" fn wasm_config_enable_nan_canonicalization(config: &mut wasm_config_t) {
```

Comment:
It might be a good idea to add a second variable (enable: true/false), in case someone want to overwrite the default value (and that default value is true)

Reply:
This function must move into the `unstable/engine.rs` module for the moment.
Also, I propose the following API: `wasm_config_nan_canonicalization(config;: &mut wasm_config_t, enable: bool)`. Thoughts?

Finally, this function should be added inside the `build.rs` script, inside the `exclude_items_from_wasm_c_api` function, so that it's not present in the `wasmer.h*` header files.